src/m2k_impl.cpp

```diff
@@ -315,6 +316,134 @@ std::vector<M2kAnalogOut*> M2kImpl::getAllAnalogOut()
 	return m_instancesAnalogOut;
 }
 
+bool M2kImpl::hasMixedSignal()
+{
+	auto digital_impl = dynamic_cast<M2kDigitalImpl*>(getDigital());
+	return digital_impl->hasRateMux();
+}
+
+void M2kImpl::startMixedSignalAcquisition(unsigned int nb_samples)
+{
+	bool hasAnalogTrigger;
+	bool hasDigitalTrigger;
+	analogSource = m_trigger->getAnalogSource();
+	digitalSource = m_trigger->getDigitalSource();
+
+	hasAnalogTrigger = this->hasAnalogTrigger();
+	hasDigitalTrigger = this->hasDigitalTrigger();
+
+	if (!hasAnalogTrigger && !hasDigitalTrigger) {
+		m_trigger->setAnalogSource(NO_SOURCE);
+		m_trigger->setDigitalSource(SRC_ANALOG_IN);
+	} else if (!hasDigitalTrigger) {
+		m_trigger->setDigitalSource(SRC_ANALOG_IN);
+	} else if (!hasAnalogTrigger) {
+		m_trigger->setAnalogSource(SRC_DIGITAL_IN);
+	}
+
+	m_instancesDigital.at(0)->setRateMux();
+
+	for (auto analogIn : m_instancesAnalogIn) {
+		analogIn->startAcquisition(nb_samples);
+	}
+	for (auto digital : m_instancesDigital) {
+		digital->startAcquisition(nb_samples);
+	}
+
+	if (!hasAnalogTrigger && !hasDigitalTrigger) {
+		m_trigger->setAnalogMode(CHANNEL_1, ALWAYS);
+		m_trigger->setAnalogSource(CHANNEL_1);
+	} else if (!hasDigitalTrigger) {
+		m_trigger->setAnalogSource(analogSource);
+	} else if (!hasAnalogTrigger) {
+		m_trigger->setDigitalSource(digitalSource);
+	}
+}
```

Comment:
You could add some comments in here how this works, `The 3 cases before starting the acquisition, the rate mux, start acquisition, the 3 cases after starting the acquisition`, but only if it makes sense for you.

Reply:
Added some comments.